src/gennylib/CMakeLists.txt

```diff
@@ -16,143 +16,16 @@
-    src/version.cpp
-)
-target_include_directories(gennylib_core
-    PUBLIC
-        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
-        $<INSTALL_INTERFACE:include>
-)
-target_link_libraries(gennylib_core
-    yaml-cpp
-    loki
-    metrics
-    value_generators
-    Boost::boost Boost::log
-    MongoCxx::mongocxx
-)
-
-# Concepts to do with Actors
-add_library(gennylib_cast STATIC
-    src/Actor.cpp
-    src/ActorProducer.cpp
-    src/Cast.cpp
-    src/ExecutionStrategy.cpp
-    src/RateLimiter.cpp
-)
-target_link_libraries(gennylib_cast gennylib_core)
-
-# Additional utilities and core concepts
-# Note that this contains both what you need to compile Actors and what you need to compile a Driver
-add_library(gennylib SHARED
-    src/version.cpp
-    src/context.cpp
-    src/Orchestrator.cpp
-    src/PoolFactory.cpp
-    )
-
-# Define headers for this library. PUBLIC headers are used for
-# compiling the library, and will be added to consumers' build
-# paths.
-target_include_directories(gennylib
-    PRIVATE
-        src
-)
-
-# Link in the sublibraries into the main library
-target_link_libraries(gennylib
-    PUBLIC
-        gennylib_cast
-        gennylib_core
-)
-if(ENABLE_JASPER)
-    target_link_libraries(gennylib
-        PRIVATE
-            jasperlib
-    )
-endif()
-
-# Link in OpenSSL
-target_link_libraries(gennylib
-    PUBLIC
-        OpenSSL::Crypto
-        OpenSSL::SSL
-)
-
-# 'make install' to the correct locations (provided by GNUInstallDirs).
-install(TARGETS  gennylib gennylib_cast gennylib_core
-        EXPORT   GennyLibraryConfig
-        ARCHIVE  DESTINATION ${CMAKE_INSTALL_LIBDIR}
-        LIBRARY  DESTINATION ${CMAKE_INSTALL_LIBDIR}
-        RUNTIME  DESTINATION ${CMAKE_INSTALL_BINDIR})  # This is for Windows
-install(DIRECTORY include/
-        DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
-        FILES_MATCHING PATTERN *.hpp)
-
-# Pull together our utils, header directories, and testing dependencies into one static target
-add_library(gennylib_testutil STATIC
-    test/ActorHelper.cpp
-    test/MongoTestFixture.cpp
-)
-target_include_directories(gennylib_testutil
-    PUBLIC
-        src
-        test
-)
-target_sources(gennylib_testutil
-    PUBLIC
-        $<TARGET_OBJECTS:catch2>
-)
-target_link_libraries(gennylib_testutil gennylib catch2_headers)
-
-add_executable(test_gennylib
-    test/Cast_test.cpp
-    test/context_test.cpp
-    test/conventions_test.cpp
-    test/ExecutionStrategy_test.cpp
-    test/PoolFactory_test.cpp
-    test/PhaseLoop_test.cpp
-    test/version_test.cpp
-    test/orchestrator_test.cpp
-
-    # Test the helper
-    test/ActorHelper_test.cpp
-)
-target_link_libraries(test_gennylib gennylib_testutil)
-
-add_executable(benchmark_gennylib
-    test/PhaseLoop_benchmark.cpp
-    test/RateLimiter_benchmark.cpp
-    test/orchestrator_benchmark.cpp
-)
-
-target_link_libraries(benchmark_gennylib gennylib_testutil)
-
-
-add_executable(test_gennylib_with_server
```

Comment:
I moved Actor tests to `cast_core/test` and I didn't create a distinction for "`_with_server`" tests - I don't think there's much value in separating out tests by target when we have catch2 tags that could facilitate isolating which tests to run.